processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -212,11 +215,21 @@ private void updateState(SubscriptionStateListener.State newState) {
 
     @Override
     public void run() {
-        updateState(SubscriptionStateListener.State.INITIALIZING);
-        consumeManager.init(subscribeTopics());
+        started = true;
+        try {
+            if (!terminated) {
```

Comment:
There's potential race such that:

* A: the subscription thread to execute `run()`
* B: the thread calling `.close()` (or it's siblings)

* B: `terminated = true`
* A: `started = true`
* A:  calls `consumer.subscribe()`
* B: do delegate completion of `loopTerminateFuture` to A

in such case, without this check the process still can be cleaned up, but because of A can be already ware of termination flag, I thought running `consumer.subscribe()` which then involves partition assignments => processors creation is not negligibly time-consuming work to let it proceed.
So I thought checking terminated at the very top of the `run()` would be worth.

Reply:
Thanks for explanation.
Makes sense.